### PR TITLE
feat(Infinity Bots): add presence

### PR DESCRIPTION
**Pull request by CodeMeAPixel**
**Describe the changes in this pull request:**

<details>
<summary> Add Infinity Bot List </summary>
<br>
1. https://master.ddoser.wtf/c146c3.png
2. https://master.ddoser.wtf/4e76c3.png
3. https://master.ddoser.wtf/1f73ce.png
</details>

NOTE: Images etc have been updated since the last pic was taken <3

### CODE REVIEW

File: websites/I/IBL/presence.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
+const presence = new Presence({
+    clientId: '815553000470478850'
+}), timestamp = Math.floor(Date.now() / 1000);
+
+presence.on('UpdateData', async () => {
+    const showTimestamp: boolean = await presence.getSetting('timestamp'),
+    showButtons: boolean = await presence.getSetting('buttons'),
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:7]: Use Promise.all and use type generics too

File: websites/I/IBL/presence.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+presence.on('UpdateData', async () => {
+    const showTimestamp: boolean = await presence.getSetting('timestamp'),
+    showButtons: boolean = await presence.getSetting('buttons'),
+    presenceData: PresenceData = {
+        largeImageKey: 'logo',
+        startTimestamp: timestamp
+    },
+
+        loc = document.location.pathname;
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:13]: Use some appropriate variable names please
```suggestion
    },
        { pathname } = document.location;
```

File: websites/I/IBL/presence.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+        if (loc === '/') {
+            presenceData.details = '👀 Viewing: Home Page';
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:16]:
```suggestion
            presenceData.details = 'Viewing homepage';

```

File: websites/I/IBL/presence.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            presenceData.state = 'Just Chilling Here!';
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:17]: no

File: websites/I/IBL/presence.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+            presenceData.smallImageKey = 'home';
+            presenceData.smallImageText = 'What do you expect?'
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:19]: no

File: websites/I/IBL/presence.ts [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            presenceData.buttons = [
+                {
+                    label: 'Visit Infinity Bots',
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:22]: rephrase this

File: websites/I/IBL/presence.ts [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+                    url: document.location.href
+                }
+            ]
+
+        } else if (loc.includes('/search')) {
+            const [query] = document.location.href.split('q=')[1].split('&');
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:28]: variable is only used once

File: websites/I/IBL/presence.ts [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+            presenceData.details = `👀 Viewing: Bot Search`;
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:29]: Searching for a bot?

File: websites/I/IBL/presence.ts [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            presenceData.state = `🔍 Query: ${query}`
+            presenceData.smallImageKey = 'search';
+            presenceData.smallImageText = 'Searching for a Bot'
```
Comment (EncryptedDev) [inline websites/I/IBL/presence.ts:32]: You don't need to repeat yourself

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
It would be much nicer if you used a object with the type`Record<string, PresenceData>` instead as this is very repetitive. 

Even if you don't, follow my requests through the cluster of else-if statements.

File: websites/I/IBL/dist/metadata.json [hunk 10]
```diff
@@ -0,0 +1,55 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.5",
```
Comment (EncryptedDev) [inline websites/I/IBL/dist/metadata.json:2]: Invalid schema

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/I/IBL/dist/metadata.json [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+      },
+      {
+        "name": "Anaxes",
+        "id": "567885938160697377"
+      },
+      {
+        "name": "Dexter",
+        "id": "780079091172900884"
+      }
+    ],
```
Comment (EncryptedDev) [inline websites/I/IBL/dist/metadata.json:21]: Surely it doesn't take four people to make a 150 line presence

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/I/IBL/dist/metadata.json [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    "description": {
+      "en": "The future of Discord Bot Listing Services!",
+      "fr": "L'avenir des services de référencement de Discord Bot !",
+      "nl": "De toekomst van Discord Bot Listing Services!"
+    },
+    "url": "infinitybots.gg",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/SSH8W2p.png",
```
Comment (EncryptedDev) [inline websites/I/IBL/dist/metadata.json:29]: must be 512x 512

**Review by EncryptedDev — DISMISSED** [review]

**Review by CodeMeAPixel — COMMENTED** [review]
Most issues have been addressed

File: websites/I/IBL/dist/metadata.json (hunk 11 above)
Comment (TheRealToxicDev) [reply websites/I/IBL/dist/metadata.json]: Adjusted to myself and Anaxes

File: websites/I/IBL/dist/metadata.json (hunk 10 above)
Comment (TheRealToxicDev) [reply websites/I/IBL/dist/metadata.json]: Updated Schema

File: websites/I/IBL/presence.ts (hunk 1 above)
Comment (TheRealToxicDev) [reply websites/I/IBL/presence.ts]: @EncryptedDev can you explain this in a little more detail please as this is the only issue i didnt really understand what you mean

**Comment by TheRealToxicDev** [comment]
Update: I’m not really sure why the lint check here is failing myself and @DiscordAnaxes have both ran lint on our end and it says it’s fine 🤔

File: websites/I/IBL/presence.ts (hunk 1 above)
Comment (EncryptedDev) [reply websites/I/IBL/presence.ts]:
```suggestion
     const [ showTimestamp,showButtons ] = await Promise.all([
     	presence.getSetting<boolean>('timestamp'),
     	presence.getSetting<boolean>('buttons'),
     ])
```

File: websites/I/IBL/dist/metadata.json (hunk 12 above)
Comment (EncryptedDev) [reply websites/I/IBL/dist/metadata.json]: This was not addressed

**Comment by EncryptedDev** [comment]
> Update: I’m not really sure why the lint check here is failing myself and @DiscordAnaxes have both ran lint on our end and it says it’s fine 🤔

Not that hard, just run the lint script

File: websites/I/InfinityBotList/dist/metadata.json [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    "id": "510065483693817867"
+  },
+  "contributors": [
+    {
+      "name": "Anaxes",
+      "id": "567885938160697377"
+    }
+  ],
+  "description": {
+    "en": "The future of Discord Bot Listing Services!",
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/dist/metadata.json:15]: Why is this not the same description matching the metadata?

Comment (EncryptedDev) [reply websites/I/InfinityBotList/dist/metadata.json:15]: image

File: websites/I/InfinityBotList/presence.ts [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+    },
+    { pathname, href } = document.location;
+
+  if (pathname === "/") presenceData.details = "👀 Viewing homepage";
+  else if (pathname.includes("/search")) {
+    presenceData.details = "👀 Searching for a bot";
+    presenceData.state = `🔍 Query: ${href.split("q=")[1].split("&")[0]}`;
+    presenceData.smallImageKey = "search";
+  } else if (pathname.endsWith("/vote")) {
+    const botName = document
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:20]: Variable is only used once

File: websites/I/InfinityBotList/presence.ts [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+      .textContent.split("#")[0];
+    presenceData.details = "🎉 Voting for a bot";
+    presenceData.state = `🤖 Bot: ${botName}`;
+    presenceData.smallImageKey = "vote";
+  } else if (pathname.includes("/list")) {
+    presenceData.details = "💻 Browsing all bots";
+    presenceData.state = `📖 Page: ${href.split("page=")[1].split("&")[0]}`;
+    presenceData.smallImageKey = "robot";
+  } else if (pathname.includes("/bots/")) {
+    const botName = document
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:31]: Variable is only used once

File: websites/I/InfinityBotList/presence.ts [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+        .textContent.split("#")[0]
+    }`;
+    presenceData.smallImageKey = "profile";
+  } else if (pathname.includes("/profile")) {
+    presenceData.details = "👀 Viewing my profile";
+    presenceData.state = "👨‍💻 Editing: Probably Nothing";
+    presenceData.smallImageKey = "profile";
+  } else if (pathname.includes("/partners")) {
+    presenceData.details = "💻 Browsing the Partners Page";
+    presenceData.state = "Just Showing some love!";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:52]: No

File: websites/I/InfinityBotList/presence.ts [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.details = "👀 Viewing a user profile";
+    presenceData.state = `👤 User: ${
+      document
+        .querySelector("body > center:nth-child(5) > h2")
+        .textContent.split("#")[0]
+    }`;
+    presenceData.smallImageKey = "profile";
+  } else if (pathname.includes("/profile")) {
+    presenceData.details = "👀 Viewing my profile";
+    presenceData.state = "👨‍💻 Editing: Probably Nothing";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:48]: No

File: websites/I/InfinityBotList/presence.ts [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.state = `📖 Page: ${href.split("page=")[1].split("&")[0]}`;
+    presenceData.smallImageKey = "robot";
+  } else if (pathname.includes("/bots/")) {
+    const botName = document
+      .querySelector("body > center:nth-child(6) > div:nth-child(2) > h1")
+      .textContent.split("#")[0];
+    presenceData.details = "👀 Viewing a bot page";
+    presenceData.state = `🤖 Bot: ${botName}`;
+    presenceData.smallImageKey = "robot";
+    presenceData.smallImageText = "Checking out a Bot!";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:37]: No

File: websites/I/InfinityBotList/presence.ts [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "partners";
+    presenceData.buttons = [
+      {
+        label: "View Partners",
+        url: href
+      }
+    ];
+  } else if (pathname === "/queue") {
+    presenceData.details = "👀 Viewing the bot queue";
+    presenceData.state = "Just looking around";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:62]: No

File: websites/I/InfinityBotList/presence.ts [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "queue";
+    presenceData.buttons = [
+      {
+        label: "View the Queue",
+        url: href
+      }
+    ];
+  } else if (pathname.includes("/about")) {
+    presenceData.details = "👀 Viewing about us";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:71]:
```suggestion
    presenceData.details = "👀 Reading about us";
```

File: websites/I/InfinityBotList/presence.ts [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.state = "📖 Reading some Boring Stuff";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:72]: No

File: websites/I/InfinityBotList/presence.ts [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "read";
+  } else if (pathname.endsWith("/panel")) {
+    presenceData.details = "👀 Viewing the Staff Panel";
+    presenceData.state = "Just Browsing some Stuff";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:76]: No

File: websites/I/InfinityBotList/presence.ts [hunk 23]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "panel";
+  } else if (pathname.endsWith("/panel/queue")) {
+    presenceData.details = "👀 Viewing the Staff Panel";
+    presenceData.state = "➕ Action: Approving Bots";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:80]:
```suggestion
    presenceData.state = "➕ Approving Bots";
```

File: websites/I/InfinityBotList/presence.ts [hunk 24]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "panel";
+  } else if (pathname.includes("/panel/certification")) {
+    presenceData.details = "👀 Viewing the Staff Panel";
+    presenceData.state = "➰ Action: Certifying Bots";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:84]:
```suggestion
    presenceData.state = "➰ Certifying Bots";
```

File: websites/I/InfinityBotList/presence.ts [hunk 25]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "panel";
+  } else if (pathname.includes("/premium")) {
+    presenceData.details = "👀 Viewing the Premium Page";
+    presenceData.state = "💻 Browsing: Plans & Pricing";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:88]: This is a given provided you're viewing the premium page...

File: websites/I/InfinityBotList/presence.ts [hunk 26]
```diff
@@ -0,0 +1,110 @@
...
+    presenceData.smallImageKey = "premium";
+    presenceData.buttons = [
+      {
+        label: "View Premium Plans",
+        url: document.location.href
+      }
+    ];
+  } else if (pathname === "/bots/certification") {
+    presenceData.details = "👀 Viewing bot certification";
+    presenceData.state = "🔱 Action: Browsing or Applying";
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:98]: use code to detect which one they are doing, or use this
```suggestion
    presenceData.state = "🔱 Browsing or Applying";
```

File: websites/I/InfinityBotList/presence.ts [hunk 27]
```diff
@@ -0,0 +1,110 @@
...
+  } else if (pathname === "/bots/add") {
+    presenceData.details = "👀 Viewing the Add Bot Page";
+    presenceData.smallImageKey = "robot";
+  }
+
+  if (!showButtons) delete presenceData.buttons;
+  if (showTimestamp) presenceData.startTimestamp = timestamp;
+
+  if (presenceData.details) presence.setActivity(presenceData);
+  else presence.setActivity();
```
Comment (EncryptedDev) [inline websites/I/InfinityBotList/presence.ts:109]:
```suggestion
  presence.setActivity();
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
State should only used to provide extra information or be used when text overflows, do not just use it because you can. Also, please remove the eyes emoji, it is freaking me out...

**Comment by EncryptedDev** [comment]
As this is a bot list, you are subject to follow the following guidelines: image

Provide proof you match these.

**Comment by TheRealToxicDev** [comment]
> As this is a bot list, you are subject to follow the following [guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md):
> 
> * Presences for services like Discord Bot/Server Lists must follow these extra requirements:
>   
>   * The domain should be at least **6 months** old.
>   * Unique visitors per day:
>     
>     * For 6 to 12 month old domains: **20,000 unique visitors/day**.
>     * For 12+ month old domains: **45,000 unique visitors/day**.
>   * The website can't be on a cheap domain like `.xyz`, `.club` and so on.
>   * The website itself must have a very good quality, design, etc.
> 
> Provide proof you match these.

Ahh yes unfortunately we don’t the GG domain is only about a month old it was a Christmas gift. Nor do we meet the unique views yet sadly.

**Comment by EncryptedDev** [comment]
You should of checked this before opening this pull request as it would of saved you and me some time.

**Pull request closed without merging**